### PR TITLE
Vidadhee/mac support

### PR DESCRIPTION
**Pull request by dadheech115**
This PR contains work for ::

1. Support of both Mac and iOS platform for this SDK.
2. Addition of .podspec file to distribute through Cocoapods

### CODE REVIEW

File: MSGraphCoreSDK/MSGraphCoreSDK/Authentication/MSAuthenticationProvider.h [hunk 1]
```diff
@@ -3,7 +3,7 @@
 //
 
 #import <Foundation/Foundation.h>
-#import <UIKit/UIKit.h>
+//#import <UIKit/UIKit.h>
```
Comment (deepak2016) [inline MSGraphCoreSDK/MSGraphCoreSDK/Authentication/MSAuthenticationProvider.h]: why commented? can you remove it?

File: MSGraphCoreSDK/MSGraphCoreSDK/Middleware/Protocols/MSGraphMiddleware.h [hunk 2]
```diff
@@ -3,7 +3,7 @@
 //
 
 #import <Foundation/Foundation.h>
-#import <UIKit/UIKit.h>
+//#import <UIKit/UIKit.h>
```
Comment (deepak2016) [inline MSGraphCoreSDK/MSGraphCoreSDK/Middleware/Protocols/MSGraphMiddleware.h]: Same here, can this be removed?

File: MSGraphCoreSDK/MSGraphCoreSDK/Authentication/MSAuthenticationProvider.h (hunk 1 above)
Comment (dadheech115) [reply MSGraphCoreSDK/MSGraphCoreSDK/Authentication/MSAuthenticationProvider.h]: Done.

File: MSGraphCoreSDK/MSGraphCoreSDK/Middleware/Protocols/MSGraphMiddleware.h (hunk 2 above)
Comment (dadheech115) [reply MSGraphCoreSDK/MSGraphCoreSDK/Middleware/Protocols/MSGraphMiddleware.h]: Done.

**Review by deepak2016 — APPROVED** [review]

**Review by NakulSabharwal — APPROVED** [review]

**Pull request merged**